Jenkinsfile

```diff
@@ -1,15 +1,28 @@
 pipeline {
-    agent {
-        label "java8"
-    }
+    agent none
+    stages {
+        stage("Validate") {
+            matrix {
+                axes {
+                    axis {
+                        name 'JAVA_VERSION'
+                        values 'java8', 'neo-java'
+                    }
+                }
+                agent {
+                    label env.JAVA_VERSION
```

Comment:
I ended up using the labels as the axis values, but that doesn't feel great.

How would I do a map from values to labels in a declarative pipeline?

Or maybe this _is_ the way to go, and it's just a matter of naming labels consistently?